proto/osmosis/poolmanager/v1beta1/swap_route.proto

```diff
@@ -38,3 +38,35 @@ message SwapAmountOutSplitRoute {
     (gogoproto.nullable) = false
   ];
 }
+
+// The following are used in creating a stateful routing graph.
+
+message Route {
+  uint64 pool_id = 1;
+  string token = 2;
+}
+
+message Routes { repeated Route routes = 1; }
+
+// The following are used to make printing parts of the structure as strings and
+// not as memory addresses. This is important in creating keys that are used to
+// determine best routes. This should NOT be used in anything stateful.
+message InnerMap { map<string, Routes> inner_map = 1; }
+
+message RoutingGraphMap { map<string, InnerMap> graph = 1; }
+
+// The following are used to serialize and deserialize the routing graph into
+// KVStore.
+message InnerMapEntry {
+  string key = 1;
+  Routes value = 2;
+}
+
+message Inner { repeated InnerMapEntry entries = 1; }
+
+message RoutingGraphEntry {
+  string key = 1;
+  Inner value = 2;
+}
+
+message RoutingGraph { repeated RoutingGraphEntry entries = 1; }
```

Comment:
From discussions,I understand that this is to be able to serialize a map of maps.

However, I would not be able to tell otherwise. Could you please elaborate on this via comments?

Reply:
Further comments added here https://github.com/osmosis-labs/osmosis/pull/6895/commits/bbbc4396f9b3e6036a2fda02ecb89968cc2d2789